### ISSUE TITLE
False positive when calling overloaded `*args` with union of tuples

### ISSUE THREAD

**Issue by randolf-scholz** [microsoft/pyright#10813]
Code sample in [pyright playground](https://pyright-play.net/?reportUnreachable=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMpgBuApiADZgCGAJgFC0ACRpFNt1xwUwYYAFACpKINAGcAXJhQwAlFAC0APigA5MCmKSAdDsbNyVOhy49%2BQkRKiiYIOUtXrNUHVvrHYxa3wAekmAFcEMmIAbVQYABopGABdKAAfWEDgkOsQKLSYu2U1DXFaKELuXkFvGVogA)

```python
from typing import overload

@overload
def foo(*args: int) -> None: ...
@overload
def foo(*args: str) -> None: ...

def test(x: tuple[int, int] | tuple[str, str]) -> None:
    foo(*x)
```

Produces false positive error:

```
Argument of type "int | str" cannot be assigned to parameter "args" of type "str" in function "foo"
  Type "int | str" is not assignable to type "str"
    "int" is not assignable to "str"  (reportArgumentType)
```
